consensus/babe/babe_test.go

```diff
@@ -435,12 +436,100 @@ func TestBabeAnnounceMessage(t *testing.T) {
 
 	for i := 0; i < int(babesession.config.EpochLength); i++ {
 		block := <-newBlocks
-		blockNumber := big.NewInt(int64(i))
+		blockNumber := big.NewInt(int64(0))
 		if !reflect.DeepEqual(block.Header.Number, blockNumber) {
 			t.Fatalf("Didn't receive the correct block: %+v\nExpected block: %+v", block.Header.Number, blockNumber)
 		}
 	}
+}
+
+func TestBuildBlock(t *testing.T) {
+	rt := newRuntime(t)
+	cfg := &SessionConfig{
+		Runtime: rt,
+	}
+
+	babesession, err := NewSession(cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = babesession.configurationFromRuntime()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// https://github.com/paritytech/substrate/blob/5420de3face1349a97eb954ae71c5b0b940c31de/core/transaction-pool/src/tests.rs#L95
+	txb := []byte{1, 212, 53, 147, 199, 21, 253, 211, 28, 97, 20, 26, 189, 4, 169, 159, 214, 130, 44, 133, 88, 133, 76, 205, 227, 154, 86, 132, 231, 165, 109, 162, 125, 142, 175, 4, 21, 22, 135, 115, 99, 38, 201, 254, 161, 126, 37, 252, 82, 135, 97, 54, 147, 201, 18, 144, 156, 178, 38, 170, 71, 148, 242, 106, 72, 69, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 216, 5, 113, 87, 87, 40, 221, 120, 247, 252, 137, 201, 74, 231, 222, 101, 85, 108, 102, 39, 31, 190, 210, 14, 215, 124, 19, 160, 180, 203, 54, 110, 167, 163, 149, 45, 12, 108, 80, 221, 65, 238, 57, 237, 199, 16, 10, 33, 185, 8, 244, 184, 243, 139, 5, 87, 252, 245, 24, 225, 37, 154, 163, 142}
```

Comment:
Just a side thought here outside of scope. It might be nice to have these examples reusable by other packages, which would make it easy to update across all packages and tests. I think we already have some duplication and some outdated examples.

Reply:
This is a fantastic idea, event within a single package this might be useful